### PR TITLE
applications: machine_learning: Switch to new configuration scheme

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

**Comment by pdunaj** [comment]
I have compared the resulting configuration files in build directory and found no differences.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -63,10 +63,18 @@ nRF9160: Asset Tracker v2
...
-* Updated information about custom build types.
+* Added:
+
+  * Non-secure configuration for building :ref:`nrf_machine_learning_app` with :ref:`zephyr:thingy53_nrf5340`.
+  * Secure configuration for building :ref:`nrf_machine_learning_app` with :ref:`zephyr:nrf5340dk_nrf5340`.
+  * Power manager to the :ref:`nrf_machine_learning_app` application.
+
+* Updated:
+
+  * Updated information about custom build types.
```
Comment (MarekPieta) [inline doc/nrf/releases/release-notes-changelog.rst:74]: Should we use word `updated` twice here?

File: applications/machine_learning/README.rst [hunk 2]
```diff
@@ -297,10 +302,11 @@ For example, the configuration files for the Thingy:52 are defined in the :file:
 
 The following configuration files can be defined for any supported board:
 
-* :file:`app_build_type.conf` - Kconfig configuration file for a build type.
+* :file:`prj_build_type.conf` - Kconfig configuration file for a build type.
   To support a given build type for the selected board, you must define the configuration file with a proper name.
-  For example, the :file:`app_ZDebug.conf` defines configuration for ``ZDebug`` build type.
-* :file:`dts.overlay` - DTS overlay file specific for the board.
+  For example, the :file:`prj_release.conf` defines configuration for ``release`` build type.
+  The :file:`prj.conf` without any suffic defines the ``debug`` build type.
```
Comment (MarekPieta) [inline applications/machine_learning/README.rst:308]: `suffix`

File: applications/machine_learning/CMakeLists.txt [hunk 3]
```diff
@@ -7,49 +7,18 @@
...
-  )
-
-# Use static partition layout to ensure consistency between builds.
-# This is to ensure settings storage will be at the same location after the DFU.
-set(PM_STATIC_YML_FILE
-  ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/pm_static_${CMAKE_BUILD_TYPE}.yml
-  )
+################################################################################
 
-# Define configuration files.
```
Comment (MarekPieta) [inline applications/machine_learning/CMakeLists.txt:49]: There are some issues with naming convention for Thingy:53 files. Cannot add comments directly to renames so I list them here.

Comment (MarekPieta) [reply applications/machine_learning/CMakeLists.txt:49]: Should we use name `prj_debug_nus.conf` or `prj_nus.conf`? nRF Desktop uses the second naming convention while ML app uses the first one currently.

Comment (MarekPieta) [reply applications/machine_learning/CMakeLists.txt:49]: The PM static configuration file for `debug_rtt` configuration is missing (there is `pm_static` file that probably should be used for that purpose). For both secure and non-secure build variant.

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/machine_learning/CMakeLists.txt (hunk 3 above)
Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: > Should we use name prj_debug_nus.conf or prj_nus.conf? nRF Desktop uses the second naming convention while ML app uses the first one currently.

The effect of scripting. That should be `nus`. I don't want to repeat `debug` everywhere.

Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: > The PM static configuration file for debug_rtt configuration is missing (there is pm_static file that probably should be used for that purpose). For both secure and non-secure build variant.

Was it missing from the start? I have not removed any files - script only renamed them

Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: btw - when there is no pm_static specific file the default pm_static will be used. This is why I left it as is.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (pdunaj) [reply doc/nrf/releases/release-notes-changelog.rst:74]: This is the copy-paste from nrf_desktop. I will leave it for @greg-fer to comment. I will fix if needed.

File: applications/machine_learning/CMakeLists.txt (hunk 3 above)
Comment (MarekPieta) [reply applications/machine_learning/CMakeLists.txt:49]: I think renaming with script handled something improperly.

Comment (MarekPieta) [reply applications/machine_learning/CMakeLists.txt:49]: We could consider defining only a single pm_static then (consistently for both variants of Thingy). Both versions are acceptable for me, but I would prefer to keep in constistent.

File: applications/machine_learning/README.rst (hunk 2 above)
Comment (pdunaj) [reply applications/machine_learning/README.rst]: fixed

File: applications/machine_learning/CMakeLists.txt (hunk 3 above)
Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: fixed

**Comment by pdunaj** [comment]
The second force-push is rebase

File: applications/machine_learning/CMakeLists.txt (hunk 3 above)
Comment (MarekPieta) [reply applications/machine_learning/CMakeLists.txt:49]: The static PM definition is still not fixed for Thingy:53 (both secure and non-secure variant)

Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: > The pm_static_rtt.yml, seems to be named pm_static

missed that :/

Comment (pdunaj) [reply applications/machine_learning/CMakeLists.txt:49]: should be fixed now

**Comment by pdunaj** [comment]
rebased

**Review by greg-fer — COMMENTED** [review]
As mentioned in #6095 , I'd rather see the general configuration docs in gs_modifying.rst updated first, and then propagated to each single application, rather than the other waya round. Also because updating gs_modifying.rst could end up in shortening the build type sections in the application readmies. @pdunaj , I'll set up a meeting about this.

**Comment by pdunaj** [comment]
> As mentioned in #6095 , I'd rather see the general configuration docs in gs_modifying.rst updated first, and then propagated to each single application, rather than the other waya round. Also because updating gs_modifying.rst could end up in shortening the build type sections in the application readmies. @pdunaj , I'll set up a meeting about this.

But can we merge this one? We would need to review all app doc anyway in the other ticket.

**Review by greg-fer — APPROVED** [review]

**Comment by greg-fer** [comment]
Unblocking because of the #6121

**Review by MarekPieta — APPROVED** [review]

**Pull request merged**